File: ConsoleApp1/ConsoleApp1/Xuanji.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApp1
{
    class Xuanji
    {
        public int x, y;
        public List<int> list;

        public Xuanji()
        {
            list = new List<int>();
        }

        public void Set(int x, int y)
        {
            this.x = x;
            this.y = y;
        }

        public void Add(int s)
        {
            list.Add(s);
        }

        public void Remove(int s)
        {
            list.Remove(s);
        }

        public int Count()
        {
            return list.Count;
        }

        public bool Contains(int s)
        {
            return list.Contains(s);
        }

    }
}

File: ConsoleApplication1/ConsoleApplication1/xuanji.cpp
#include "stdafx.h"
#include "xuanji.h"



int x;
int y;
int l[9];

xuanji::xuanji()
{

}

void xuanji::set(int a,int b)
{
	this->x = a;
	this->y = b;
	for (int i = 0; i < 9; i++)
	{
		l[i] = 1;
	}
}

bool xuanji::remove(int x)
{
	if (l[x - 1] == 1)
	{
		l[x - 1] = 0;
		return true;
	}
	else return false;
}

void xuanji::add(int x)
{
	l[x - 1] = 1;
}

int xuanji::length()
{
	int len = 0;
	for (int k = 0; k < 9; k++)
	{
		if (l[k] == 1) len++;
	}
	return len;
}

int xuanji::choose(int b)
{
	for (int i = 0; i<9; i++)
	{
		if (l[i] == 1) b--;
		else continue;
		if (b == 0) return i + 1;
	}
}




